library/std/src/io/error.rs

```diff
@@ -127,6 +133,35 @@ pub enum ErrorKind {
     /// requested to not occur.
     #[stable(feature = "rust1", since = "1.0.0")]
     WouldBlock,
+    /// A filesystem object is, unexpectedly, not a directory.
+    ///
+    /// For example, a filesystem path was specified where one of the intermediate directory
+    /// components was, in fact, a plain file.
+    #[stable(feature = "io_error_unix_more", since = "1.51.0")]
+    NotADirectory,
```

Comment:
This looks like window's [`ERROR_DIRECTORY`](https://docs.microsoft.com/en-us/windows/win32/debug/system-error-codes--0-499-)